dronecan/sensors/rc/1140.RCInput.uavcan

```diff
@@ -1,9 +1,27 @@
-uint8 STATUS_QUALITY_VALID = 1 # quality field is valid
+#
+# RC channels
+#
+# The quality field can carry different information on the quality of a radio link, as determined by the flags in the status bitmask:
+# - STATUS_QUALITY_VALID: The quality field is valid and represents RSSI in scaled units, 0 is no signal, 255 is "full" signal. The
+#   mapping of a link's internal RSSI representation onto the scaled units is device dependent. A common mapping can be
+#     RSSI >= -50 dB: 255
+#     RSSI > -120 dB and RSSI < -50 dBm: 255 / 70 * (RSSI + 120) but not smaller than 1
+#     RSSI <= -120 dB: 1
+# - STATUS_QUALITY_LQ: The quality field is valid and represents LQ in percent, 0 is no signal, 100 is 100% link quality
+# - STATUS_QUALITY_RSSI_DBM: The quality field is valid and represents RSSI in inverted dBm, 0 is no signal, 1 = -1 dBm, 255 = -255 dBm
+# - STATUS_QUALITY_SNR: The quality field is valid and represents SNR in device dependent units, value is shifted by 128 to allow
+#   representing positive and negative values: 128 = 0, 255 = +127, 1 = -127, 0 is no signal
+# Only one of these flags can be set at any time.
+
+uint8 STATUS_QUALITY_VALID = 1 # quality field is valid and represents RSSI in scaled uints, 0 is no signal, 255 is "full" signal
 uint8 STATUS_FAILSAFE = 2 # receiver has lost contact with transmitter
+uint8 STATUS_QUALITY_LQ = 4 # quality field is valid and represents LQ in percent, 0 is no signal, 100 is 100% link quality
+uint8 STATUS_QUALITY_RSSI_DBM = 8 # rssi in inverted dBm, 0 is no signal, 1 = -1 dBm, 255 = -255 dBm
+uint8 STATUS_QUALITY_SNR = 16 # quality field is valid and represents SNR in device dependent units, shifted by 128
```

Comment:
The intent I gathered from the discussion was that this be a three bit field, with eight possibilities. Something like

```
uint8 STATUS_QUALITY_KIND = 28
uint8 STATUS_QUALITY_KIND_QUALITY = 0
uint8 STATUS_QUALITY_KIND_LQ = 4
uint8 STATUS_QUALITY_KIND_RSSI_DBM = 8
uint8 STATUS_QUALITY_KIND_SNR = 12
```

Therefore the code can write `if ((status & STATUS_QUALITY_KIND) == STATUS_QUALITY_KIND_SNR) { ... }`, we make better use of available bits, and avoid being able to encode impossible combinations.

My personal preference would be that the valid flag is true only if the quality field represents quality. That way older code will just ignore the different meanings rather than cycling between them. Though the older code may flicker between "unavailable" and the correct value, maybe worth exploring.

Reply:
if to have individual or "compressed" flags is mostly matter of taste, so no comment from my side

the suggested method IS however backwards compatible, i.e. older code will work as expected ... as much as possible with that approach at all. I don't see that any other use of flags would have any difference in this regards.

Reply:
changed as adviced